effekt/jvm/src/main/scala/effekt/Runner.scala

```diff
@@ -186,18 +226,21 @@ object MLRunner extends Runner[String] {
     if canRunExecutable("mlton") then Right(())
     else Left("Cannot find mlton. This is required to use the ML backend.")
 
-  /**
-   * Compile the LLVM source file (`<...>.ll`) to an executable
-   *
-   * Requires LLVM and GCC to be installed on the machine.
-   * Assumes [[path]] has the format "SOMEPATH.ll".
-   */
-  def eval(path: String)(using C: Context): Unit =
+  override def build(path: String)(using C: Context): String = 
     val out = C.config.outputPath()
     val buildFile = (out / "main.mlb").canonicalPath
     val executable = (out / "mlton-main").canonicalPath
     exec("mlton",
       "-default-type", "int64", // to avoid integer overflows
       "-output", executable, buildFile)
-    exec(executable)
+    executable
+
+  /**
+   * Compile the MLton source file (`<...>.sml`) to an executable.
+   *
+   * Requires the MLton compiler to be installed on the machine.
+   * Assumes [[path]] has the format "SOMEPATH.sml".
+   */
+  def eval(path: String)(using C: Context): Unit =
+    exec(build(path))
```

Comment:
This is now actually always the same implementation and could be moved to the parent trait.

Reply:
The documentation should be moved to another reasonable place (like `compile`, or `build`)